### ISSUE TITLE
Rewinding and forwarding broken on Amazon Prime video

### ISSUE THREAD

**Issue by ShrirajHegde** [polywock/globalSpeed#57]
Rewinding and forwarding are broken (recently) on amazon prime video

Behavior:  

For the first usage on the page load only rewind works, and rewinding after skipping intro doesn't work but rewinding already played video and forwarding works fine

 (**Essentially only part of the video which is already played once works**)

By doesn't work, I mean the jump to the duration happens, after that video gets "stuck" (not paused) which can be fixed by pausing and unpausing.

**Comment by polywock** [comment]
Thanks for the feedback. I don't think that issue is recent since I've encountered it a month back. I've tried finding a workaround back then as well but gave up. Prime Video uses a custom player like Netflix and requires a special way to rewind/forward. For Netflix. I've added a workaround as documentation about their player API was documented online. 

For Prime Video, I can't find any documentation. The source code is minified and hard to parse. I've tried reverse engineering their player API. To rewind 5 seconds I think I'll have to do something like this... 

`f.playbackStore.seek(f.timelineInfoStore.getTimelineInfo().positionMs - 5000))`

But, the `f` Player instance thingy seems to only exist inside of a function. I don't think it's exposed globally like for Netflix. I'll probably wait until someone finds a way to do it.

For now I just use the arrow keys for Prime video. I've also added a URL rule to the seek hotkeys to disable them on Amazon.

**Comment by ShrirajHegde** [comment]
is it possible to fix it in a hacky way by toggling the pause/play button since that works manually

**Comment by polywock** [comment]
Prime Video seeking is working fine for me right now so I can't test it. But, maybe try a system like this... 

![z](https://user-images.githubusercontent.com/31208859/107126468-27554600-687e-11eb-95b7-8b9826b89b67.png)

Z would rewind, pause, and unpause in sequence.

**Comment by ShrirajHegde** [comment]
I am sorry, I couldn't figure out how to setup like above

**Comment by polywock** [comment]
Maybe this video would help... 

https://user-images.githubusercontent.com/31208859/107138959-cfeac080-68e5-11eb-8a27-dd22fb78a0e6.mov

You could also send me your config information and I'll set it up. You can export/copy the config file in the bottom of the options page.  

![z](https://user-images.githubusercontent.com/31208859/107139043-3f60b000-68e6-11eb-8b21-c828c73407f1.png)

**Comment by ShrirajHegde** [comment]
AMAZING !!!!!!!!


Works like charm. For me it doesn't even matter if I don't create URL rules, the pause toggle is not noticeable


![image](https://user-images.githubusercontent.com/66936172/107539176-b6e65600-6bea-11eb-9ef4-3f4e3c57bab5.png)

**Comment by ShrirajHegde** [comment]
sure, I will use it as regular for few days to check for any problems.

**Comment by ShrirajHegde** [comment]
This works well, one small change I would suggest is to toggle instead of on and off to enable seeking while the video is paused without accidentally unpausing it.

The screenshot I have shown above works seamlessly.

**Comment by ShrirajHegde** [comment]
> Great, thanks for the suggesting this hack. Let me know if ends up not working. I'll probably include it in the next update as the default behavior for Amazon URLs.

Apparently, for some Global Speed reason, MS Edge works fine as intended without any workaround.

Malfunctions on firefox, brave just not (Chromium) MS edge for some reason.

**Comment by polywock** [comment]
An update, I've implemented your hack as the default behavior on Amazon sites. If you're on the latest version (2.7.2), you can remove the pause hotkeys.

Thanks again for the contribution.
